### PR TITLE
[VMEC] Update dependency versions

### PR DESCRIPTION
**Pull request by benjaminfaber**
Some of the issues with previous dependencies have been fixed, no longer need to fix to old versions.

### CODE REVIEW

File: V/VMEC/build_tarballs.jl [hunk 1]
```diff
@@ -65,11 +65,10 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     # MbedTLS is an indirect dependency, fix the version for building 
-    BuildDependency(PackageSpec(name = "MbedTLS_jll", version = v"2.24.0")),
-    Dependency("MPICH_jll"; platforms=filter(!Sys.iswindows, platforms)),
-    Dependency("OpenBLAS_jll"),
+    BuildDependency(PackageSpec(name = "MbedTLS_jll")),
```
Comment (giordano) [inline V/VMEC/build_tarballs.jl:68]: libblastrampoline works well only with Julia v1.7+, and there is also the problem that libblastrampoline 3 (the one used in Julia v1.7) and 5 (the one you're using to build here) have incompatible ABIs: https://github.com/JuliaRegistries/General/pull/64542#issuecomment-1189195017

File: V/VMEC/build_tarballs.jl [hunk 2]
```diff
@@ -65,11 +65,10 @@ products = [
...
+    Dependency("MPItrampoline_jll"; platforms=filter(!Sys.iswindows, platforms)),
```
Comment (giordano) [inline V/VMEC/build_tarballs.jl:69]: For MPITrampoline you probably want to follow the example of the other packages using it.

Comment (benjaminfaber) [reply V/VMEC/build_tarballs.jl]: I followed the examples of the other packages using MPItrampoline and it seems to build correctly.  If you approve, could this get merged soon?  This is intended to resolve MPI difficulties that have been occurring on HPC systems.

File: V/VMEC/build_tarballs.jl (hunk 1 above)
Comment (benjaminfaber) [reply V/VMEC/build_tarballs.jl:57]: Fixed to work with 1.7, I saw an example in [P/Pfapack](https://github.com/JuliaPackaging/Yggdrasil/P/Pfapack/build_tarballs.jl) of how to do this for 1.7 and 1.8, should I be following that?

Comment (benjaminfaber) [reply V/VMEC/build_tarballs.jl:99]: I removed the libblastrampoline dependence for now.

**Comment by giordano** [comment]
I think you're missing something, because the platforms don't include the MPI tag

**Comment by benjaminfaber** [comment]
@giordano I think I have fixed all the issues and am successfully building with MPItrampoline appropriately, I would appreciate a review

**Pull request merged**